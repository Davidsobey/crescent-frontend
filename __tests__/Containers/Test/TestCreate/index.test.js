import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import TestCreate from '../../../../src/Containers/Test/TestCreate/index.js';


const mockStore = configureStore();
const initialState = {
  CourseReducer:{
    courses: [],
    loading: false,
    newCourseId: 1,
  },
  ModuleReducer:{
    modules: [],
    loading: false,
    newModuleId: 1,
  },
  TestReducer:{
    tests: [],
    loading: false,
    creating: false,
    openRedirectModal: false,
  },
};
const store = mockStore(initialState);

describe('<TestCreate />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<TestCreate store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});