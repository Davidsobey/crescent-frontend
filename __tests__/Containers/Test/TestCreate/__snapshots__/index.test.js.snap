// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestCreate /> render() renders the component 1`] = `
<Connect(Form(TestCreate))
  courses={Array []}
  courses_loading={false}
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="testCreate"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  modules={Array []}
  modules_loading={false}
  newCourseId={1}
  newModuleId={1}
  openRedirectModal={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  test_creating={false}
  tests={Array []}
  tests_loading={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
